### PR TITLE
Adds a new positive quirk, "Spacer Born". 

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Adds a new 7 point positive quirk, "Spacer Born". Totally not inspired by The Expanse, don't look at the branch name. 

You were born in space, rather than on a planet, so your physiology has adapted differently.
You are more comfortable in space, and way less comfortable on a planet. 

Benefits:
   - You are slightly taller. (No mechanical effect)
   - You take 20% less damage from pressure damage.
   - You take 20% less damage from cold environments. 
   - You move 10% faster while floating (NOT drifting, this is zero gravity movement while beside a wall). 
   - You drift 20% faster (Drifting through zero gravity, untethered to anything)
   - While in space (z-level-wise, not turf wise), you lose some disgust overtime. 
      - While experiencing no-gravity for an extended period of time, you will start regenerating stamina and reduce stuns at a very low rate.  
   - If you are assigned to shaft miner (or the map is Icebox), you are awarded with a 25% wage bonus (hazard pay). 

Downsides:
   - While on a planet (Yes, this includes Icebox and planetary maps), you gain gravity sickness:
      - Passive accrue disgust (slightly lessened on Icebox) (Capped at low levels)
      - Choking, after extended periods (disabled on Icebox)
      - Slower movement 
      - Weaker stamina (disabled on Icebox)
      - Suffocation from extended periods (disabled on Icebox) (Lungs aren't adapted)
      - Mood debuff

(Effects not final)

## Why It's Good For The Game

I'd figure I throw my hat in with the Positive Quirk Curse. 

This is a quirk that improves your ability in a niche circumstance (low gravity / dangerous pressure), with some downsides that are only generally in effect if you play a few roles (or it's Icebox). 

Because of this I think it'll provide an interesting niche, where Spacer Born engineers are slightly better than their counterparts due to their origin (moving faster in space without a jetpack, withstanding pressure). However, at the same time, if the mining outpost sustains damage and needs repairs... suddenly your buff over your cohorts disappears, and you have to brave somewhere hostile to your body. 

Ultimately, the goal of the quirk is to encourage people to approach situations a bit differently. 
Or take it as a challenge and play shaft miner. 

## Changelog

:cl: Melbert
add: Adds a new 7 point positive quirk, "Spacer Born". You were born in space, and as a result your body's adapted to life in artificial gravity, making you much more effective and comfortable in lower gravity. However, travelling planet-side is quite a chore, especially if you're assigned to work there. 
add: Adds a chemical: Ondansetron, created by Oil + Nitrogen + Oxygen + Ethanol catalyst. A powerful Antiemetic (lowers disgust).
/:cl:

### CODE REVIEW

**Comment by SomeoneYouProbablyKnow** [comment]
If suffering on icebox is what it takes to get from 5'11 to 6'0 so be it.

File: code/datums/quirks/positive_quirks.dm [hunk 1]
```diff
@@ -362,3 +362,165 @@
...
+	// Yes it's assumed, for planetary maps, that you start at gravity sickness.
+
+/// Proc to check quirk holder moves to determine when we should start tracking
+/datum/quirk/spacer_born/proc/arrived_at_station(mob/living/source, atom/old_loc)
+	SIGNAL_HANDLER
+
+	var/new_z = source.loc?.z
+	if(is_station_level(new_z) || is_mining_level(new_z))
+		UnregisterSignal(source, COMSIG_MOVABLE_MOVED)
+		start_tracking()
```
Comment (LemonInTheDark) [inline code/datums/quirks/positive_quirks.dm:421]: why do you use Moved for this? it feels like you have this backward

File: code/datums/quirks/positive_quirks.dm [hunk 2]
```diff
@@ -362,3 +362,165 @@
...
+	UnregisterSignal(quirk_holder, COMSIG_MOVABLE_MOVED)
+
+	if(!QDELING(quirk_holder))
+		quirk_holder.clear_mood_event("spacer")
+		quirk_holder.remove_movespeed_modifier(/datum/movespeed_modifier/spacer)
+		quirk_holder.remove_status_effect(/datum/status_effect/spacer)
+
+		var/mob/living/carbon/human/human_quirker = quirk_holder
+		human_quirker.set_mob_height(HUMAN_HEIGHT_MEDIUM)
+		human_quirker.physiology.pressure_mod /= 0.8
```
Comment (LemonInTheDark) [inline code/datums/quirks/positive_quirks.dm:407]: early return

File: code/datums/quirks/positive_quirks.dm [hunk 3]
```diff
@@ -362,3 +362,165 @@
...
+
+	if(skip_timers)
+		on_planet_for_too_long(afflicted, TRUE)
+	else
+		var/exercise_bonus = afflicted.has_status_effect(/datum/status_effect/exercised) ? 2 : 1
+		planetside_timer = addtimer(CALLBACK(src, PROC_REF(on_planet_for_too_long), afflicted), planet_period * exercise_bonus, TIMER_UNIQUE | TIMER_STOPPABLE)
+		afflicted.add_mood_event("spacer", /datum/mood_event/spacer/on_planet)
+		afflicted.add_movespeed_modifier(/datum/movespeed_modifier/spacer/on_planet)
+		afflicted.remove_status_effect(/datum/status_effect/spacer) // removes the wellness effect.
+		to_chat(afflicted, span_danger("You feel a bit sick under the gravity here."))
```
Comment (LemonInTheDark) [inline code/datums/quirks/positive_quirks.dm:472]: similar

File: code/datums/quirks/positive_quirks.dm [hunk 4]
```diff
@@ -362,3 +362,165 @@
...
+	if(planetside_timer)
+		deltimer(planetside_timer)
+		planetside_timer = null
+
+	if(skip_timers)
+		comfortably_in_space(afflicted, TRUE)
+	else
+		recovering_timer = addtimer(CALLBACK(src, PROC_REF(comfortably_in_space), afflicted), recover_period, TIMER_UNIQUE | TIMER_STOPPABLE)
+		afflicted.remove_status_effect(/datum/status_effect/spacer) // removes the sickness effect, but not the other modifiers.
+		to_chat(afflicted, span_green("You start feeling better now that you're back in space."))
```
Comment (LemonInTheDark) [inline code/datums/quirks/positive_quirks.dm:509]: here too

File: code/datums/status_effects/debuffs/spacer.dm [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+/datum/status_effect/spacer/gravity_wellness/tick(seconds_per_tick, times_fired)
+	var/in_nograv = !owner.has_gravity()
+	var/nograv_mod = in_nograv ? 1 : 0.5
+	owner.adjust_disgust(-1 * disgust_healing_per_tick * nograv_mod)
+
+	if(in_nograv)
+		seconds_in_nograv += (initial(tick_interval) * 0.1)
+	else
+		seconds_in_nograv = 0
+		return
```
Comment (LemonInTheDark) [inline code/datums/status_effects/debuffs/spacer.dm:38]:
```suggestion
	if(!in_nograv)
		seconds_in_nograv = 0
		return
	seconds_in_nograv += (initial(tick_interval) * 0.1)
	```
```

File: code/datums/status_effects/debuffs/spacer.dm [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+/atom/movable/screen/alert/status_effect/gravity_sickness
+	name = "Gravity Sickness"
+	desc = "The gravity of the planet around you is making you feel sick and tired."
+	icon_state = "paralysis"
+
+/datum/mood_event/spacer
+	category = "spacer"
+
+/datum/mood_event/spacer/in_space
+	description = "Space is long and dark and empty, but it's my home."
```
Comment (LemonInTheDark) [inline code/datums/status_effects/debuffs/spacer.dm:84]: I love filk

File: code/datums/quirks/positive_quirks.dm [hunk 7]
```diff
@@ -362,3 +362,165 @@
...
+ * * skip_timers - if TRUE, this is being done instantly / should not have feedback (such as in init)
+ */
+/datum/quirk/spacer_born/proc/comfortably_in_space(mob/living/afflicted, skip_timers = FALSE)
+	if(QDELETED(src) || QDELETED(afflicted))
+		return
+
+	recovering_timer = null
+	afflicted.apply_status_effect(/datum/status_effect/spacer/gravity_wellness)
+	afflicted.add_mood_event("spacer", /datum/mood_event/spacer/in_space)
+	afflicted.add_movespeed_modifier(/datum/movespeed_modifier/spacer/in_space)
```
Comment (LemonInTheDark) [inline code/datums/quirks/positive_quirks.dm:524]: Do you think this should modify drift speed?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/datums/quirks/positive_quirks.dm (hunk 7 above)
Comment (LemonInTheDark) [reply code/datums/quirks/positive_quirks.dm:524]: I guess just the quirk itself not space acclimating or anything

Comment (MrMelbert) [reply code/datums/quirks/positive_quirks.dm:524]: I did want it to speed up drifting but I didn't look into how, guess I should do that

File: code/datums/quirks/positive_quirks.dm (hunk 1 above)
Comment (MrMelbert) [reply code/datums/quirks/positive_quirks.dm:421]: Figured moved would result in it starting tracking as soon as possible vs in gameplay, it doesn't necessary need to check for planet stuff every time the mob moved (so moved z does the job)

**Comment by HoodMilk** [comment]
Too high cost for something that doesn't have many positive effects imo. The faster movement is the only notable one and is neat roundstart, but jetpacks exist and can be printed by robotics. On the other hand the negative effects are bad enough that people will just swap off of this quirk when it's icebox or when they play miner because there's literally no reason to keep it there.

I'd say just remove the negatives and keep the high point cost, or just make the quirk neutral since there are positives and negatives to it.

If you wanna go really spicy you could split it into two quirks and make the planetary weakness a negative quirk that gives exactly the same amount of points that the quirk with the buffs costs (Or less, if the buffs are seen as more powerful than the weaknesses).

**Comment by timothymtorres** [comment]
Cool idea but I think you are trying to add too many effects for one quirk.

**Comment by necromanceranne** [comment]
I have a bit of an issue with the idea of the quirk being overly negative on specifically one map. This wouldn't be so bad if our maps were largely space stations only, since having negatives on only lavaland is pretty minor and I think is a fine downside.

But generating disgust constantly while on icebox could probably get really, really harsh. Especially given the 7 point cost. There are no upsides on icebox, its only negatives, since pressure loss is quite unusual on icebox given the terrestrial atmosphere typically doesn't allow areas to completely vacuum under most station destruction circumstances.

Maybe on those maps, these characters have some kind of item that uses a charge that counteracts the disgust? I'm not sure. Or maybe a hazard pay to their paycheck for unideal circumstances.

Overall maybe if it was a bit of a lower cost it wouldn't be too harsh to carry it into icebox.

**Comment by PotatoOfChaos** [comment]
a shame that simple character customisation options like height have to be relegated to gameplay affecting perks,  but other than that minor complaint, it's nice i guess, the float speed is something really unique

**Comment by Jacquerel** [comment]
> >Choking, after extended periods (disabled on Icebox)
> 
> Just choking? Regardless of internals? This should have some valid reason.

Their lungs are collapsing under high Gs, they don't have the strong bones or muscles to support breathing there.

**Comment by KittyNoodle** [comment]
Why don’t gravity generators make them queasy

**Comment by itseasytosee** [comment]
Pretty cool, but heres an obligatory:
"Stamina regeneration is antithetical to the current stamina system and will either be useless (if too low) or disturb the entire rickety balance structure (if too high) that is stamina based combat"
Thank you for your time.

**Comment by zxaber** [comment]
> > Choking, after extended periods (disabled on Icebox)
> 
> Just choking? Regardless of internals? This should have some valid reason.

Choking is probably better than crumpling into a ball and taking massive brute damage when your stamina wears out (though that would be mechanically cooler).

**Comment by Cheshify** [comment]
Why is this part of the quirk system? It feels very strange that roleplayers need to take a bunch of negative quirks just so they can roleplay being a spacer.

This could be the interesting start of an origin system independent from quirks and species.

**Comment by Jacquerel** [comment]
> Why is this part of the quirk system? It feels very strange that roleplayers need to take a bunch of negative quirks just so they can roleplay being a spacer.

If you don't want it to have mechanical backing you can just say "I'm a spacer", that was already an option.

**Comment by MrMelbert** [comment]
I have added some effects to make it a bit more appealing to Icebox (cold resistance).

> Pretty cool, but heres an obligatory:

Yes I'm aware, the number is pretty low and is mostly intended for flavor, at most maybe will protect you from 1 disabler shot. About the same as ephedrine (won't let you tank a baton) and entirely requires you be in zero gravity for a while.

**Comment by Thunder12345** [comment]
Don't see the point of giving miners a pay bonus when it's extremely unlikely anybody is going to want to go mining when they get slowed to half speed after being planetside for just a few minutes.

**Comment by EricZilla** [comment]
I think they should suffer more on icebox

**Comment by MrMelbert** [comment]
> Don't see the point of giving miners a pay bonus when it's extremely unlikely anybody is going to want to go mining when they get slowed to half speed after being planetside for just a few minutes.

Yeah well I don't see the point in your comment that offers no alternatives so myeh

**Comment by CRITAWAKETS** [comment]
maybe give access to some sort of Buck Rogers-like grav belt atleast on icebox which needs charging. So you end up losing a pretty important equipment slot in the process aswell as needing to charge an item after a while, but it's an option if you don't want the debuffs.

File: code/datums/quirks/positive_quirks.dm [hunk 8]
```diff
@@ -362,3 +362,221 @@
 
 /datum/quirk/item_quirk/signer/remove()
 	qdel(quirk_holder.GetComponent(/datum/component/sign_language))
+
+/datum/quirk/spacer_born
```
Comment (san7890) [inline code/datums/quirks/positive_quirks.dm:366]: this is a shit load of code (200+ lines!) just being appended onto this file i think it's time we start splitting it out (starting now). files are free

File: code/datums/quirks/positive_quirks.dm [hunk 9]
```diff
@@ -362,3 +362,221 @@
...
+/**
+ * Used to check if we should start or stop timers based on the quirk holder's location.
+ *
+ * * afflicted - the mob arriving / same as quirk holder
+ * * skip_timers - if TRUE, this is being done instantly / should not have feedback (such as in init)
+ */
+/datum/quirk/spacer_born/proc/check_z(mob/living/source, skip_timers = FALSE)
+	var/turf/open/new_turf = get_turf(source)
+
+	// What is a planet?
```
Comment (san7890) [inline code/datums/quirks/positive_quirks.dm:474]: if you're going to have sixteen lines of documentation to help narrow it down for a generic(ish) edge case, why not  make this a more generic global proc that can be used elsewhere in the code?

**Review by san7890 — CHANGES_REQUESTED** [review]
mainly nits regarding code organization

should this be incompatible with other quirks like wheelchair bound? feels silly to be so rough-and-ready about being in space and get nauseous from moving around when you're literally sitting down most of the time

also also, what happens if someone with this quirk gets dwarfism mutation? i've had too many polluted icon caches breaking shit with carbon icon generation so i'd appreciate this being tested rather than "i think the code will be fine". ultimately whatever, just something that popped in my mind

**Comment by san7890** [comment]
> Why don’t gravity generators make them queasy

it's a psychological thing that manifests physiologically. boom problem solved

**Comment by MrMelbert** [comment]
> what happens if someone with this quirk gets dwarfism mutation

`TRAIT_DWARF` has priority over current set mob height, so it wins

File: code/datums/quirks/positive_quirks.dm (hunk 8 above)
Comment (MrMelbert) [reply code/datums/quirks/positive_quirks.dm:366]: this is gonna be the fourth time we've moved and renamed quirk files

File: code/modules/reagents/reagent_containers/patch.dm [hunk 10]
```diff
@@ -45,6 +45,12 @@
 	list_reagents = list(/datum/reagent/medicine/c2/synthflesh = 20)
 	icon_state = "bandaid_both"
 
+/obj/item/reagent_containers/pill/patch/ondansetron
+	name = "ondansetron patch"
+	desc = "Alleviates nausia. May cause drowsiness."
```
Comment (san7890) [inline code/modules/reagents/reagent_containers/patch.dm:50]:
```suggestion
	desc = "Alleviates nausea. May cause drowsiness."
```

whoops wish i caught this one the first time

**Review by san7890 — CHANGES_REQUESTED** [review]

**Comment by AlamoTurtle** [comment]
> I have a bit of an issue with the idea of the quirk being overly negative on specifically one map. This wouldn't be so bad if our maps were largely space stations only, since having negatives on only lavaland is pretty minor and I think is a fine downside.
> 
> But generating disgust constantly while on icebox could probably get really, really harsh. Especially given the 7 point cost. There are no upsides on icebox, its only negatives, since pressure loss is quite unusual on icebox given the terrestrial atmosphere typically doesn't allow areas to completely vacuum under most station destruction circumstances.
> 
> Maybe on those maps, these characters have some kind of item that uses battery charge that counteracts the disgust? I'm not sure. Or maybe a hazard pay to their paycheck for unideal circumstances. Maybe both.
> 
> Overall maybe if it was a bit of a lower cost it wouldn't be too harsh to carry it into icebox.

Gonna counter this with a "they get higher pay on icebox so they should buy some sol dry every now and again with their extra cash". I get it's a straight up debuff on icebox, so why the 7 points invested, but it can work as a challenge on top of all the struggles of icebox. Maybe they don't suffer from the thinner atmosphere of icebox so no need for internals (lower oxygen threshold) and higher thresholds for "your lungs freezing", being so suited to lower temperatures, so that you won't need to crack out the old internals box every time some space asshole airlock cards departures open and lets some moon wind in past the firelocks.

**Comment by LemonInTheDark** [comment]
I don't think it's bad for the quirk to be worse on icebox. It being not that would fail to sell its concept, and rob players of the feeling they signed up for.
I like the hazard pay and cold resistance. Doesn't apply to everyone but it's important to realize that sometimes feeling bad is an experience, and trying to prevent that ruins the vibes and really the fun of something.

File: code/__HELPERS/levels.dm [hunk 11]
```diff
@@ -17,3 +17,40 @@
...
+ */
+/proc/is_on_a_planet(atom/what)
+	ASSERT(!isarea(what))
+
+	var/turf/open/new_turf = get_turf(what)
+	if(isnull(new_turf))
+		return FALSE
+
+	var/z_level_checks = (is_station_level(new_turf.z) && SSmapping.config.planetary) || is_mining_level(new_turf.z)
+	var/planetary_check = istype(new_turf) && new_turf.planetary_atmos
```
Comment (LemonInTheDark) [inline code/__HELPERS/levels.dm:55]: These names suck what are you checking FOR

File: code/__HELPERS/levels.dm [hunk 12]
```diff
@@ -17,3 +17,40 @@
...
+	return (z_level_checks || planetary_check) && new_turf.has_gravity()
```
Comment (LemonInTheDark) [inline code/__HELPERS/levels.dm:56]: should early return off has_gravity(), it's a common case too so it's not a massive extra cost here

File: code/datums/quirks/positive_quirks/spacer.dm [hunk 13]
```diff
@@ -0,0 +1,195 @@
...
+
+/// Check on Z change whether we should start or stop timers
+/datum/quirk/spacer_born/proc/spacer_moved(mob/living/source, turf/old_turf, turf/new_turf, same_z_layer)
+	SIGNAL_HANDLER
+
+	check_z(source)
+	// Using Z moved because we don't urgently need to check on every single turf movement for planetary status.
+	// If you've arrived at a "planet", the entire Z is gonna be a "planet".
+	// It won't really make sense to walk 3 feet and then suddenly gain / lose gravity sickness.
+	// If I'm proven wrong, swap this to use Moved.
```
Comment (LemonInTheDark) [inline code/datums/quirks/positive_quirks/spacer.dm:98]: this comment location is weird

File: code/__HELPERS/levels.dm (hunk 11 above)
Comment (LemonInTheDark) [inline code/__HELPERS/levels.dm:55]: if you don't think planetary atmos will matter why do we check for it at all? if this is a per z level sort of thing why even check?

File: code/__HELPERS/levels.dm (hunk 12 above)
Comment (LemonInTheDark) [inline code/__HELPERS/levels.dm:56]: similar for has_gravity actually

File: code/datums/quirks/positive_quirks/spacer.dm [hunk 14]
```diff
@@ -0,0 +1,195 @@
...
+	human_quirker.physiology.cold_mod /= 0.8
+
+/datum/quirk/spacer_born/proc/start_tracking(mob/living/tracking)
+	var/new_z = tracking.loc?.z
+	// Some maps may decide to spawn people on mining Z, so just cast a wide net.
+	if(!is_station_level(new_z) && !is_mining_level(new_z) && !is_centcom_level(new_z))
+		return FALSE
+
+	UnregisterSignal(quirk_holder, COMSIG_MOVABLE_Z_CHANGED) // gets rid of arrived_at_station registration if present
+	RegisterSignal(quirk_holder, COMSIG_MOVABLE_Z_CHANGED, PROC_REF(spacer_moved)) // replaces it with spacer_moved
```
Comment (LemonInTheDark) [inline code/datums/quirks/positive_quirks/spacer.dm:78]: why do we not just register onto spacer_moved at the start? aren't the checks in this proc identical to what's inside is_on_a_planet (and thus spacer_moved)

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/__HELPERS/levels.dm (hunk 12 above)
Comment (LemonInTheDark) [reply code/__HELPERS/levels.dm:56]: actually why is this here in the first place, see below

**Comment by YesterdaysPromise** [comment]
Lore-wise, every moth would be forced to take it since they live in space flotilla. At this point, why not make an one-hard-select origin system separate to quirks?

**Comment by TwinIndifferent** [comment]
Cool and all, but 7 points cost for having mediocre buffs while gravity generator is offline/EVA and having debuffs on planet surface is... weird? 90% of time players with such quirk will almost never walk on Lavaland, effectively negating the downside **but** the upsides are niche 90% of the time too. And Icebox is not 50/50.

To me, there is supposed to be middle ground - considering that low-G creates weak bones, spacer is supposed to be weaker in jobs requiring physical strength of an earth-grown man. Something like -15% to pull speed and weakier aggressive grab strength.
For positive part, having vacuum resistance is weird and niche too - you either never meet vacuum long enough to be in a serious danger, or you are already dead. Having lower O2 consumption would go better instead of vacuum resist - air is precious on the Belt after all, and on Earth we have ethnical groups in high elevations capable to breathe with lower pressure too!

**Comment by CRITAWAKETS** [comment]
make them have a negative mood reaction to being in "space wind" (excited atmos turfs) if it doesn't take too much performance, since having wind in a space habitat is generally not a good thing

File: code/__HELPERS/levels.dm (hunk 11 above)
Comment (MrMelbert) [reply code/__HELPERS/levels.dm:55]: Primarily for away missions or other edge cases

**Comment by LemonInTheDark** [comment]
> make them have a negative mood reaction to being in "space wind" (excited atmos turfs) if it doesn't take too much performance, since having wind in a space habitat is generally not a good sign

This is not a good metric, don't use it.

**Comment by ATH1909** [comment]
should mention Ondansetron in the opening post

**Comment by timothymtorres** [comment]
None of the current quirks have a dozen separate effects.  You also are mixing negative effects into something that should be *only* have a positive effect.  All of these effects should be split into individual quirks.

<details>
<summary>Quirk List for Reference</summary>

Trait_menu

</details>

**Comment by necromanceranne** [comment]
> None of the current quirks have a dozen separate effects. You also are mixing negative effects into something that should be _only_ have a positive effect. All of these effects should be split into individual quirks.
> Quirk List for Reference

That's not entirely true. Some things have a mixture, just not to such an extent as this quirk. It doesn't fit elegantly into positive, neutral or negative, but I don't think your suggestion is functional either. This is ultimately meant to majorly change your functional circumstances in a drastic way, and making the elements of the quirk piecemeal ruins that entirely.

**Comment by zxaber** [comment]
Prosthetic Limb is also half-n-half. It makes EMPs hurt a lot, but you get an easy method to self-heal with wires and welders due to how health works - often it's enough to break out of slow-limp movespeed.

This quirk is cool and thematic.

**Comment by san7890** [comment]
i do not think there is a way to split this one quirk into multiple quirks  (i.e. atomizing it into height being a positive quirk and nausea on planets being a negative seems silly, etc.) without losing something special when it is all a sum of its parts. it's just unclean, messy, and it allows people to just "be tall" by taking a alcoholism or something- there's no blend that really lends height value.

**Comment by EricZilla** [comment]
Since its sometimes good sometimes bad wouldn't making it a little cheaper make sense? Like, 5ish?

**Comment by SomeoneYouProbablyKnow** [comment]
The problem with lowering its cost is that the downsides are all planetside, which you should avoid by no picking miner and swapping characters on icebox. So it's a VERY tangible benefit for a very tangible cost

**Pull request merged**